components/butterfly/src/rumor.rs

```diff
@@ -333,6 +388,46 @@ mod storage {
             }
             result
         }
+
+        pub fn keys_rsr(&self) -> Vec<RumorKey> {
+            self.list
+                .read()
+                .values()
+                .flat_map(HashMap::values)
+                .map(RumorKey::from)
+                .collect()
+        }
+    }
+
+    impl<T: Transient> RumorStore<T> {
+        pub fn insert_transient_rsw(&self, rumor: T) -> bool {
```

Comment:
Also, it might be worth doing some thinking about how we can avoid `Transient` rumors being accidentally inserted via `insert_rsw` and vice versa. I'm not sure about the best way off the top of my head, but I'm pretty sure it's worth making safe via types.

Reply:
That's a good point.  I'll think on this a bit.